File: 171-excel-sheet-column-number/171-excel-sheet-column-number.cpp
class Solution {
public:
  int titleToNumber(string s) {
    	return accumulate(s.begin(), s.end(), 0, [](const int&a, const char&b){return 26 * a + (b - 'A') + 1;});

  }
};

/*
A - 26^0 * 1
B - 26^0 * 2
AA - 26^1 * 1 + 26^0 * 1 
BA - 26^1 * 2 + 26^0 * 2
AAA - 26^3 * 1 + 26^2 * 1 + 26^0 * 0
*/

File: 171-excel-sheet-column-number/171-excel-sheet-column-number.java
class Solution {
    public int titleToNumber(String columnTitle) {
        int res=0, offset = 'A'-1, l = columnTitle.length(), charCount = 26;
    for(int i=0;i<l;i++)
      res += Math.pow(charCount, i) * (columnTitle.charAt(l-i-1) - offset);
    return res;
    }
}